datastream-server/src/main/java/com/linkedin/datastream/server/assignment/BroadcastStrategy.java

```diff
@@ -59,17 +60,27 @@ public BroadcastStrategy(Optional<Integer> maxTasks) {
 
     Map<String, Set<DatastreamTask>> newAssignment = new HashMap<>();
 
+    Set<DatastreamTask> tasksAvailableToReuse = new HashSet<>();
     // Make a copy of the current assignment, since the strategy modifies it during calculation
     Map<String, Set<DatastreamTask>> currentAssignmentCopy = new HashMap<>(currentAssignment.size());
-    currentAssignment.forEach((k, v) -> currentAssignmentCopy.put(k, new HashSet<>(v)));
+    currentAssignment.forEach((k, v) -> {
+      currentAssignmentCopy.put(k, new HashSet<>(v));
+      // building the full list and then removing the reused tasks rather than building the list from dead tasks
+      // to avoid assignment of same task from dead instance to two live instances in case of interrupt exception.
+      tasksAvailableToReuse.addAll(v);
+    });
 
     for (String instance : instances) {
       newAssignment.put(instance, new HashSet<>());
       currentAssignmentCopy.putIfAbsent(instance, new HashSet<>());
```

Comment:
Not your fault: this isn't very efficient; `new HashSet` isn't needed if `instance` is not absent. This could be improved to:
```java
 currentAssignmentCopy.computeIfAbsent(instance, i -> new HashSet<>());
```
If we do that, we could also improve line 76 by avoiding the extra `get()`:
```java
Set<DatastreamTask> instanceTasks = currentAssignmentCopy.computeIfAbsent(instance, i -> new HashSet<>());
tasksAvailableToReuse.removeAll(instanceTasks);
```

Reply:
Done.